src/System.Private.CoreLib/shared/System/Globalization/TimeSpanFormat.cs

```diff
@@ -35,66 +37,53 @@ private static unsafe void AppendNonNegativeInt32(StringBuilder sb, int n, int d
         internal static readonly FormatLiterals PositiveInvariantFormatLiterals = TimeSpanFormat.FormatLiterals.InitInvariant(isNegative: false);
         internal static readonly FormatLiterals NegativeInvariantFormatLiterals = TimeSpanFormat.FormatLiterals.InitInvariant(isNegative: true);
 
-        internal enum Pattern
-        {
-            None = 0,
-            Minimum = 1,
-            Full = 2,
-        }
 
         /// <summary>Main method called from TimeSpan.ToString.</summary>
-        internal static string Format(TimeSpan value, string format, IFormatProvider formatProvider) =>
-            StringBuilderCache.GetStringAndRelease(FormatToBuilder(value, format, formatProvider));
+        internal static string Format(TimeSpan value, string format, IFormatProvider formatProvider)
+        {
+            return IsFormatC(format) ? // special-case to optimize the default TimeSpan format
+                FormatC(value) : // formatProvider ignored, as "c" is invariant
+                StringBuilderCache.GetStringAndRelease(FormatToBuilder(value, format, formatProvider));
+        }
 
         /// <summary>Main method called from TimeSpan.TryFormat.</summary>
         internal static bool TryFormat(TimeSpan value, Span<char> destination, out int charsWritten, ReadOnlySpan<char> format, IFormatProvider formatProvider)
         {
+            if (IsFormatC(format)) // special-case to optimize the default TimeSpan format
+            {
+                return TryFormatC(value, destination, out charsWritten); // formatProvider ignored, as "c" is invariant
+            }
+
             StringBuilder sb = FormatToBuilder(value, format, formatProvider);
+
             if (sb.Length <= destination.Length)
             {
-                charsWritten = sb.Length;
                 sb.CopyTo(0, destination, sb.Length);
+                charsWritten = sb.Length;
```

Comment:
I am curious, why re-order these lines?

Reply:
No great reason, just that since it's trivial to do either way, seemed better to only write the number of chars once they'd actually already been written.  It doesn't really matter.